src/libraries/System.Private.Xml/src/System/Xml/Schema/XmlSchema.cs

```diff
@@ -58,16 +58,19 @@ public class XmlSchema : XmlSchemaObject
 
         public static XmlSchema? Read(TextReader reader, ValidationEventHandler? validationEventHandler)
         {
+            ArgumentNullException.ThrowIfNull(reader);
```

Comment:
What was the behavior before the check was added? NullReferenceException?

Reply:
NullRef for all

Reply:
"Root element is missing." for `XPathDocument..ctor(TextReader)` when TextReader is null. That's just purely incorrect

Reply:
seems it was also `XmlSchema.Writ`e and there is testcase which is testing that. I'll go ahead and change that behavior since this error doesn't seem particularly useful:

```
Unhandled exception. System.InvalidOperationException: There was an error generating the XML document.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Xml.Serialization.XmlSerializationWriter.WriteStartDocument()
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationWriterXmlSchema.Write64_schema(Object o)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Xml.Serialization.XmlSerializer.Serialize(XmlWriter xmlWriter, Object o, XmlSerializerNamespaces namespaces, String encodingStyle, String id)
   at System.Xml.Schema.XmlSchema.Write(XmlWriter writer, XmlNamespaceManager namespaceManager)
   at ConsoleApp98.Program.Main(String[] args) in C:\Users\kwicher\source\repos\XmlSchemaWriteRepro\XmlSchemaWriteRepro\Program.cs:line 13
```